Changelog.md

```diff
@@ -2,6 +2,9 @@
 
 ### SDFormat 6.X.X (20XX-XX-XX)
 
+1. Move recursiveSameTypeUniqueNames from ign.cc to parser.cc and make public.
+    * [Pull request ]()
```

Comment:
reminder to update this before release

Reply:
I was about to forget about it. Thanks Ian, done.